### PR TITLE
Bug 1167913 - Add Oauth credential lookup and export process to RTD

### PR DESCRIPTION
**Pull request by mjzffr**
@wlach - as promised.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/559)

### CODE REVIEW

**Comment by tojonmz** [comment]
Thanks Maja! :) Curious was there an existing bug against which this PR/commit can be referenced? I did a quick check and found docs bug [1069293](https://bugzilla.mozilla.org/show_bug.cgi?id=1069293) but it didn't seem to refer to RTD directly.

I can enter one and hook it up to the PR if needed.

Thanks for helping out!

**Comment by mjzffr** [comment]
No bug that I know of. sorry. @wlach just mentioned that I should feel free to improve docs here and there as I work with treeherder.

File: docs/common_tasks.rst [hunk 1]
```diff
@@ -51,6 +51,26 @@ Running the tests
...
+  .. code-block:: bash
+
+      > ./manage.py export_project_credentials
+
+Here's how to look up the credentials for ``mozilla-central``, for example:
+
+  .. code-block:: python
+
+      from treeherder.etl.oauth_utils import OAuthCredentials
+      credentials = OAuthCredentials.get_credentials('mozilla-central')
```
Comment (wlach) [inline docs/common_tasks.rst]: Could you go into more detail on how to actually export the keys to a file for usage with something like TreeherderClient? I assume this just gives details on a data structure.

**Comment by wlach** [comment]
Yeah, the process for getting a change (including doc changes) into Treeherder is to file a bug in the tree management component. In this case:

https://bugzilla.mozilla.org/enter_bug.cgi?product=Tree%20Management&component=Treeherder%3A%20Docs%20%26%20Development

Then attach the PR to the bug as follows: http://globau.wordpress.com/2013/10/21/github-pull-requests-and-bugzilla/

This is a bit time consuming, I know. :/ Maybe in the future we should consider just accepting PR's for documentation without bothering with bugzilla. But for now this is how it is.

**Comment by tojonmz** [comment]
No probs, I've just added a marker bug for it, and updated the PR subject. Yes, as far as I understand it helps visibility for PM, project tracking, metrics, work completed, since everyone seems to use Bugzilla for that. At least for now.

I'm totally happy to add a bug when needed, if it helps keep the contributions flowing :)

**Comment by edmorley** [comment]
> This is a bit time consuming, I know. :/ Maybe in the future we should consider just accepting PR's for documentation without bothering with bugzilla. But for now this is how it is.

For docs changes, where a multi-line commit message is used (if appropriate; wasn't needed for this PR, since it was self-explanatory) I'm fine with not filing a bug if there wasn't one open already.

File: docs/common_tasks.rst [hunk 2]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+
+To submit data to an existing project, you need corresponding OAuth credentials, even if you're submitting to your local server.
+
+Prerequisite: you must have previously exported credentials for all data source projects. This is part of the usual Treeherder setup process.
+
+  .. code-block:: bash
+
+      > ./manage.py export_project_credentials
+
+The above command writes all project credentials to ``treeherder/etl/data/
```
Comment (mjzffr) [inline docs/common_tasks.rst]: @wlach I added some more detail up here. Is this what you were going for?

File: docs/common_tasks.rst [hunk 3]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+credentials.json`` for use by the Treeherder service.
```
Comment (wlach) [inline docs/common_tasks.rst]: Could you move this up to the previous line? I think this is an unintentional line break.

And yes, this is great. I didn't know how to get the oauth credentials directly before and now I do. :)

File: docs/common_tasks.rst [hunk 4]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+
+If you are implementing data submission with ``TreeherderClient`` and testing
+that locally, it is useful to work with your own copy of ``credentials.json``.
+You can export the credentials wherever you need with the ``--destination``
+option.
```
Comment (wlach) [inline docs/common_tasks.rst]: option. -> option:

File: docs/common_tasks.rst [hunk 5]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+Look up credentials for a project
+---------------------------------
+
+To submit data to an existing project, you need corresponding OAuth credentials, even if you're submitting to your local server.
+
+Prerequisite: you must have previously exported credentials for all data source projects. This is part of the usual Treeherder setup process.
+
+  .. code-block:: bash
+
+      > ./manage.py export_project_credentials
```
Comment (wlach) [inline docs/common_tasks.rst]: The usual practice here is to specify the prompt so it's clear these commands need to be run inside the virtual environment. In this case you'd write:

(venv)vagrant@precise32:~/treeherder$ ./manage.py export_project_credentials

File: docs/common_tasks.rst [hunk 6]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+that locally, it is useful to work with your own copy of ``credentials.json``.
+You can export the credentials wherever you need with the ``--destination``
+option.
+
+  .. code-block:: bash
+
+      > ./manage.py export_project_credentials --destination /some/path
+
+Within Treeherder, you can look up the credentials for a project like ``mozilla
+-centrals`` as follows:
```
Comment (wlach) [inline docs/common_tasks.rst]: Again, this should be one line, I think. Also ``mozilla-centrals`` -> ``mozilla-central``

File: docs/common_tasks.rst [hunk 7]
```diff
@@ -51,6 +51,42 @@ Running the tests
...
+Prerequisite: you must have previously exported credentials for all data source projects. This is part of the usual Treeherder setup process.
+
+  .. code-block:: bash
+
+      > ./manage.py export_project_credentials
+
+The above command writes all project credentials to ``treeherder/etl/data/
+credentials.json`` for use by the Treeherder service.
+
+If you are implementing data submission with ``TreeherderClient`` and testing
```
Comment (wlach) [inline docs/common_tasks.rst]: Let's link to the submitting data document inline: 

    data submission -> :doc:`data submission <submitting_data>`

**Comment by wlach** [comment]
This is great! Just a few small changes and then I think we're good to go.

**Comment by mjzffr** [comment]
Awesome, thanks for the feedback. I think I've addressed everything. 

@wlach Good thing you asked for more detail because it forced me to realize what `export_project_credentials` actually does. :)

File: docs/common_tasks.rst [hunk 8]
```diff
@@ -51,6 +51,45 @@ Running the tests
 
      (venv)vagrant@precise32:~/treeherder$ flake8
 
+
+Look up credentials for a project
+---------------------------------
+
+To submit data to an existing project, you need corresponding OAuth
+ credentials, even if you're submitting to your local server.
```
Comment (wlach) [inline docs/common_tasks.rst]: There's a stray space at the beginning of this line, which is messing a bit with formatting.

File: docs/common_tasks.rst [hunk 9]
```diff
@@ -51,6 +51,45 @@ Running the tests
...
+
+      (venv)vagrant@precise32:~/treeherder$ ./manage.py export_project_credentials
+
+The above command writes all project credentials to
+``treeherder/etl/data/credentials.json`` for use by the Treeherder service.
+
+If you are implementing :doc:`data submission <submitting_data>` with
+``TreeherderClient`` and testing that locally, it is useful to work with your
+own copy of ``credentials.json``.
+You can export the credentials wherever you need with the ``--destination``
```
Comment (wlach) [inline docs/common_tasks.rst]: May as well move this up to the previous line.

**Comment by wlach** [comment]
Oops, sorry, forgot something important in the last review. Could you fix those two last nits? I'll merge this after that.

**Comment by mjzffr** [comment]
Done.

**Comment by wlach** [comment]
Thanks!!

**Pull request merged**